src/gui/map/map_editor.cpp

```diff
@@ -3901,6 +3911,318 @@ void MapEditorController::createSymbolWidget(QWidget* parent)
 	}
 }
 
+void MapEditorController::exportOgrClicked()
+{
+	const auto& georef = map->getGeoreferencing();
+
+	// Check if map is georeferenced
+	if (georef.getState() == Georeferencing::Local)
+	{
+		QMessageBox::warning(window, tr("No Georeferencing"), tr("The map is not georeferenced.  Cannot export."));
+		return;
+	}
+
+	// Make sure GDAL can work with the georeferencing info
+	OGRSpatialReference o_spat_ref;
+	if (o_spat_ref.importFromProj4(georef.getProjectedCRSSpec().toLatin1().constData()) != OGRERR_NONE)
+	{
+		QMessageBox::warning(window, tr("Error Exporting"), tr("Failed to properly create the export georeferencing info"));
+		return;
+	}
+
+	// Make sure an object is selected
+	if (map->getNumSelectedObjects() == 0) {
+		QMessageBox::warning(window, tr("Nothing selected"), tr("No objects are selected.  Not exporting anything."));
+		return;
+	}
+
+	QSettings settings;
+	QString export_directory = settings.value(QString::fromLatin1("exportFileDirectory"), QDir::homePath()).toString();
+
+	QString filename = QFileDialog::getSaveFileName(
+						   window,
+						   tr("Export GPX, KML, or SHP file"),
+						   export_directory,
+						   tr("GPS Exchange Format (*.gpx);;Keyhole Markup Language (*.kml);;Shapefile (*.shp)"));
```

Comment:
This will probaly not be need with OgrFileExport. Anyway, both strings are not well suited for translation because they are going to change frequently: with every new format we use from GDAL.